test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/TransactionTests.cs

```diff
@@ -3,35 +3,271 @@
+            Assert.Equal(expectedDiagnostics.Length, operationOutcome.Issue.Count);
+
+            for (int iter = 0; iter < operationOutcome.Issue.Count; iter++)
+            {
+                Assert.Equal(expectedCodeType[iter], operationOutcome.Issue[iter].Code);
+                Assert.Equal(OperationOutcome.IssueSeverity.Error, operationOutcome.Issue[iter].Severity);
+                Assert.Equal(expectedDiagnostics[iter], operationOutcome.Issue[iter].Diagnostics);
+            }
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenABundleWithForeignReferenceInResourceBody_WhenSubmittingATransaction_ThenReferenceShouldNotBeResolvedAndProcessShouldSucceed()
+        {
+            var requestBundle = Samples.GetJsonSample("Bundle-TransactionWithForeignReferenceInResourceBody");
+
+            var fhirResponse = await Client.PostBundleAsync(requestBundle.ToPoco<Bundle>());
+
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+
+            foreach (var entry in fhirResponse.Resource.Entry)
+            {
+                IEnumerable<ResourceReference> references = entry.Resource.GetAllChildren<ResourceReference>();
+                foreach (var reference in references)
+                {
+                    // Asserting the conditional reference value before resolution
+                    Assert.Equal("urn:uuid:4a089b8a-b0a0-46a9-92da-c8b653aa2e73", reference.Reference);
+                }
+            }
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenATransactionBundleReferencesInResourceBody_WhenSuccessfulExecution_ReferencesAreResolvedCorrectlyAsync()
+        {
+            // Insert a resource that has a predefined identifier.
+            var resource = Samples.GetJsonSample("PatientWithMinimalData");
+            await Client.CreateAsync(resource.ToPoco<Patient>());
+
+            var bundleWithConditionalReference = Samples.GetJsonSample("Bundle-TransactionWithReferenceInResourceBody");
+
+            var bundle = bundleWithConditionalReference.ToPoco<Bundle>();
+
+            FhirResponse<Bundle> fhirResponseForReferenceResolution = await Client.PostBundleAsync(bundle);
+
+            Assert.NotNull(fhirResponseForReferenceResolution);
+            Assert.Equal(HttpStatusCode.OK, fhirResponseForReferenceResolution.StatusCode);
+
+            foreach (var entry in fhirResponseForReferenceResolution.Resource.Entry)
+            {
+                IEnumerable<ResourceReference> references = entry.Resource.GetAllChildren<ResourceReference>();
+
+                foreach (var reference in references)
+                {
+                    // Asserting the conditional reference value before resolution
+                    Assert.True(reference.Reference.Contains("/", System.StringComparison.Ordinal));
```

Comment:
Should this be a starts with since the following reference from the bundle contains a `/`? `"reference": "Patient?identifier=http:/example.org/fhir/ids|234234"`